app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -528,4 +537,15 @@ def image_tenants(image)
     end
     tenants
   end
+
+  def find_scan_info(vm_or_template_hash, ems_ref)
+    scan_time = vm_or_template_hash.select { |vm_or_template| vm_or_template["ems_ref"] == ems_ref }
+    scan_time[0]["last_scan_on"] unless scan_time.empty?
+  end
+
+  private
+
+  def last_scan_hash
+    VmOrTemplate.select(:ems_ref, :last_scan_on).where(:ems_id => collector.manager.id)
```

Comment:
lets do

```
VmOrTemplate.select(:ems_ref, :last_scan_on).where(:ems_id => collector.manager.id).where.not(:last_scan_on => nil)
```

Reply:
also use `.index_by` to turn this into hash `{"ems_ref" => "last_scan_on"}, then the lookup will be O(1)

Reply:
and memoize it here